### DISCUSSION TITLE
How to show hidden files by default?

### DISCUSSION THREAD

**Discussion by bennypowers** [nvim-neo-tree/neo-tree.nvim#353] [Q&A] [answered]
Hey!

I like the hidden files feature, but I'd like to toggle them on by default.

In other words, I always find myself hitting `H` (or `I`, then saying "D'oh!", then hitting `H`) every time I open neo-tree. I'd like the hidden files to be shown automatically, but I'd still like them to render as 'hidden', and for `H` to toggle them off.

Should I just make an autocmd to toggle them on when I enter? I think I'd end up losing the toggle state then when i move back and forth...

**Comment by nyngwang** [comment, marked as answer]
```lua
require('neo-tree').setup {
  filesystem = {
    filtered_items = {
      visible = true, -- This is what you want: If you set this to `true`, all "hide" just mean "dimmed out"
      hide_dotfiles = false,
      hide_gitignored = true,
    },
    -- unrelated
```

**Reply by bennypowers** [reply]
_nice_
Thanks!

**Reply by Zaccal** [reply]
Where is it need write? In the init.lub ?

**Reply by nyngwang** [reply]
@Zaccal I meant neo-tree.nvim's `setup { ... }`. Apologies for my shorthand-style answer, I've edited the code section.

**Reply by Zaccal** [reply]
ok thanks

**Reply by prescient-slime** [reply]
Hello, guys! I'm new to AstroNvim. Where is Neo-tree.nvim at? I'm having the same problem.

**Reply by cseickel** [reply]
Hi @prescient-slime, I'm not positive how to apply this config in AstroNvim, you'll have to ask over at that project's repo. You may be able to figure it out from their documentation: https://astronvim.github.io/Configuration/basic_configuration

**Reply by SirSharanChakradhar** [reply]
what file I must edit and add the code? what is the file name and location?

**Reply by MatthiasEckhart** [reply]
> what file I must edit and add the code? what is the file name and location?

Setup your user configuration as described in the [AstroNvim documentation](https://astronvim.github.io/Configuration/basic_configuration).

After that, just apply the changes as suggested by @nyngwang:
```bash
$ nvim ~/.config/nvim/lua/user/init.lua
```

```
  -- Configure plugins
  plugins = {
    init = {
      -- .......
    },
    -- .......
    ["neo-tree"] = {
      filesystem = {
        filtered_items = {
          visible = true, 
          hide_dotfiles = false,
          hide_gitignored = true,
        },
      }
    }
},

```

**Reply by sbouaked** [reply]
Thanks @MatthiasEckhart , I did this, but there is no changes in the hidden item list :/ 
Is there some other change / manip to do ?

I try to change all the init.lua files to be sure...

<img width="750" alt="CleanShot 2023-02-16 at 15 05 09@2x" src="https://user-images.githubusercontent.com/3003831/219386472-c5f73605-4cb0-415c-b3f5-38bacaf2108b.png">

**Reply by cjkvdev** [reply]
> Thanks @MatthiasEckhart , I did this, but there is no changes in the hidden item list :/ Is there some other change / manip to do ?

Honestly, I can say the same as well. _Wish the opts were better documented?_ Maybe I'm not looking in the right direction for my very brief introduction into the whole `neovim` world this morning. Thank you, lovely companion of a plugin.

EDIT: I'm using LazyVim, personally. Have a codeblock as shown:

```lua
    {
      "nvim-neo-tree/neo-tree.nvim",
      cmd = "Neotree",
      keys = {
        {
          "<leader>fe",
          function()
            require("neo-tree.command").execute({ toggle = true, dir = require("lazyvim.util").get_root() })
          end,
          desc = "Explorer NeoTree (root dir)",
        },
        {
          "<leader>fE",
          function()
            require("neo-tree.command").execute({ toggle = true, dir = vim.loop.cwd() })
          end,
          desc = "Explorer NeoTree (cwd)",
        },
        { "<leader>e", "<leader>fe", desc = "Explorer NeoTree (root dir)", remap = true },
        { "<leader>E", "<leader>fE", desc = "Explorer NeoTree (cwd)",      remap = true },
      },
      deactivate = function()
        vim.cmd([[Neotree close]])
      end,
      opts = {
        filesystem = {
          bind_to_cwd = false,
          follow_current_file = true,
          filtered_items = {
            visible = true,
            hide_dotfiles = false,
            hide_gitignored = false,
            hide_by_name = {
              ".DS_Store",
              "thumbs.db",
              "node_modules",
            },
          },
        },
        window = {
          mappings = {
            ["<space>"] = "none",
          },
        },
      },
```

**Reply by Jesseekoh** [reply]
@cjkvdev what file to I add this to? I'm also using Lazyvim but can't get it to work

**Reply by MatthiasEckhart** [reply]
@Jesseekoh 

If you are using [`lazy.nvim`](https://github.com/folke/lazy.nvim), something like this could work:

```
$ nvim ~/.config/nvim/lua/user/plugins/neo-tree.lua
return {
  "nvim-neo-tree/neo-tree.nvim",
  opts = {
    filesystem = {
      filtered_items = {
	 visible = true,
	 show_hidden_count = true,
	 hide_dotfiles = false,
	 hide_gitignored = true,
	 hide_by_name = {
	   -- '.git',
	   -- '.DS_Store',
	   -- 'thumbs.db',
	 },
	never_show = {},
      },
    }
  }
}
```

Obviously, the config needs to be adapted to fit your needs (check the [quickstart guide](https://github.com/nvim-neo-tree/neo-tree.nvim#quickstart)).

**Reply by Jesseekoh** [reply]
@MatthiasEckhart  thanks a lot. It worked

**Reply by carlos-ssh** [reply]
@cjkvdev
That configuration that you shared was helpful. Thanks! I can display my hidden files in the tree files.🚀🚀🚀

**Reply by domnantas** [reply]
No need to replace all the options, this seems to be enough
```lua
return {
  "nvim-neo-tree/neo-tree.nvim",
  opts = {
    filesystem = {
      filtered_items = {
        visible = true,
      },
    },
  },
}
```

**Reply by vdegenne** [reply]
OP wanted to show hidden files but hide files defined in `.gitignore`, your code will just show everything..
Also you may want to always hide some files like `.git` by default. It's always good to see all options at once.

For people webdev'ing I find this setup to be the most accurate:
```lua
return {
  "neo-tree.nvim",
  opts = {
    filesystem = {
      filtered_items = {
        visible = false, -- hide filtered items on open
        hide_gitignored = true,
        hide_dotfiles = false,
        hide_by_name = {
          ".github",
          ".gitignore",
          "package-lock.json",
          ".changeset",
          ".prettierrc.json",
        },
        never_show = { ".git" },
      },
    },
  },
}
```

**Reply by spike014** [reply]
If you are using [lazy.nvim](https://github.com/folke/lazy.nvim), something like this could work:
```
$ nvim ~/.config/nvim/lua/plugins/neo-tree.lua
return {
  "nvim-neo-tree/neo-tree.nvim",
  opts = {
    filesystem = {
      filtered_items = {
	 visible = true,
	 show_hidden_count = true,
	 hide_dotfiles = false,
	 hide_gitignored = true,
	 hide_by_name = {
	   -- '.git',
	   -- '.DS_Store',
	   -- 'thumbs.db',
	 },
	never_show = {},
      },
    }
  }
}
```
lazy.nvim will load the plugins under the `~/.config/nvim/lua/plugins`

**Reply by ahyalfan** [reply]
Thank you very much

**Reply by greycodee** [reply]
> If you are using [lazy.nvim](https://github.com/folke/lazy.nvim), something like this could work:
> 
> ```
> $ nvim ~/.config/nvim/lua/plugins/neo-tree.lua
> return {
>   "nvim-neo-tree/neo-tree.nvim",
>   opts = {
>     filesystem = {
>       filtered_items = {
> 	 visible = true,
> 	 show_hidden_count = true,
> 	 hide_dotfiles = false,
> 	 hide_gitignored = true,
> 	 hide_by_name = {
> 	   -- '.git',
> 	   -- '.DS_Store',
> 	   -- 'thumbs.db',
> 	 },
> 	never_show = {},
>       },
>     }
>   }
> }
> ```
> 
> lazy.nvim will load the plugins under the `~/.config/nvim/lua/plugins`

Thank you very much

**Comment by gldtn** [comment]
Can someone point me to the right direction? All .dotfiles  are still being hidden, I'm completely new to Nvim and slowly trying to configure it.

This is what I have on my neotree.lua file:
```lua
return {
    "nvim-neo-tree/neo-tree.nvim",
    branch = "v3.x",
    dependencies = {
        "nvim-lua/plenary.nvim",
        "nvim-tree/nvim-web-devicons", 
        "MunifTanjim/nui.nvim",
    },
    opts = {
        filesystem = {
            filtered_items = {
				hide_dotfiles = false,
                visible = true,
            },
        },
    },
    config = function(_, opts)
		require("neo-tree").setup(opts)
        vim.keymap.set("n", "<C-b>", "<CMD>Neotree toggle<CR>", {})
    end
}
```

**This has been edited 10/23/24, the code above should work!**

Thank you!

**Comment by pysan3** [comment]
@gldtn welcome onboard. You need `hide_dotfiles = false` here as literally the name suggests.

**Reply by gldtn** [reply]
Seems like my keymap line within the neo-tree.lua keeps the `opts` from working correctly, most likely from my lack of knowledge with Lua.

Once I moved it to a separate keymaps.lua the `opts` from this thread started working beautifully ;), so I removed this from my neo-tree.lua:

```lua
config = function()
        vim.keymap.set("n", "<C-b>", "<CMD>Neotree toggle<CR>", {})
end
```

Thanks for the help @pysan3

**Comment by rikkarth** [comment]
Working `neo-tree.lua` configuration for `lazyvim` with key-map configured within the configuration file.

## neo-tree.lua
```lua
return {
    "nvim-neo-tree/neo-tree.nvim",
    branch = "v3.x",
    dependencies = {
        "nvim-lua/plenary.nvim",
        "nvim-tree/nvim-web-devicons", -- not strictly required, but recommended
        "MunifTanjim/nui.nvim",
        -- "3rd/image.nvim", -- Optional image support in preview window: See `# Preview Mode` for more information
    },
    config = function()
        require("neo-tree").setup({
            filesystem = {
                filtered_items = {
                    visible = true,
                    show_hidden_count = true,
                    hide_dotfiles = false,
                    hide_gitignored = true,
                    hide_by_name = {
                        -- add extension names you want to explicitly exclude
                        -- '.git',
                        -- '.DS_Store',
                        -- 'thumbs.db',
                    },
                    never_show = {},
                },
            },
        })
        vim.keymap.set("n", "<C-n>", ":Neotree filesystem reveal left<CR>", {})
    end,
}
```

**Comment by Rayaa144** [comment]
alternatively you could toggle show hidden files by pressing shift + T

**Comment by ExploitEngineer** [comment]
I am also having the same problem my dot files are not showing in neo-tree. I have setup the lazy vim but by default i am getting this kind of neovim-tree as in the image.
I have check my plugins there is no plugin is installed name "neo-tree.nvim" and when i installed it and configure it i am getting both of them how do i fix this problem ?
![Screenshot_08-Jul_18-42-53_13763](https://github.com/user-attachments/assets/ec2622ff-41ae-4973-ad4f-cfc27e94da77)

**Reply by bashmyhed** [reply]
shift h is the default toggle key for hidden files

**Reply by gldtn** [reply]
This is "Explorer" from snacks.nvim: https://github.com/folke/snacks.nvim/blob/main/docs/explorer.md, you will need to configure the picker to show hidden files by default: https://github.com/folke/snacks.nvim/blob/main/docs/picker.md#explorer

**Reply by ExploitEngineer** [reply]
Thanks this is working...
